controller/resource_roles.go

```diff
@@ -33,19 +33,14 @@ func NewResourceRolesController(service *goa.Service, app application.Applicatio
 
 // ListAssigned runs the list action.
 func (c *ResourceRolesController) ListAssigned(ctx *app.ListAssignedResourceRolesContext) error {
-
-	var roles []role.IdentityRole
-
-	err := c.app.ResourceRepository().CheckExists(ctx, ctx.ResourceID)
+	currentIdentity, err := login.LoadContextIdentityIfNotDeprovisioned(ctx, c.app)
 	if err != nil {
-		log.Error(ctx, map[string]interface{}{
-			"resource_id": ctx.ResourceID,
-			"err":         err,
-		}, "does not exist")
-		return jsonapi.JSONErrorResponse(ctx, errors.NewNotFoundError("resource_id", ctx.ResourceID))
+		return jsonapi.JSONErrorResponse(ctx, err)
 	}
 
-	roles, err = c.app.RoleManagementService().ListByResource(ctx, ctx.ResourceID)
+	var roles []role.IdentityRole
+
+	roles, err = c.app.RoleManagementService().ListByResource(ctx, currentIdentity.ID, ctx.ResourceID)
```

Comment:
Do we need this change for this PR? I'm banking on this (unchanged) API to easily validate the migration of collaborators :)  

I was using a service account token to call this API, but this change would now need a user token.

Reply:
It's much easier to keep consistency if it's part of this PR. It also won't be very easy to extract this change to another PR. All relevant changed were made in the service and many tests.
I would prefer another tiny PR with support of migration SA here which we will revert after migration.

Reply:
I understand, sounds good. I'll make the PR after we merge this.